### DISCUSSION TITLE
Private/Public network of IPFS

### DISCUSSION THREAD

**Discussion by farhoud** [functionland/box#206] [Ideas]
In Our initial design box runs 2 instance of fula-api one is private and another is public.
Are we still on the same path.
@gitaaron and @masih If i remember has some doubt about it.
So lets discuss it here.
@keyvan-m-sadeghi @ehsan6sha @mehdibalouchi @ruffiano89 @ghorbani-m

**Comment by ehsan6sha** [comment]
@farhoud I am thinking that what is the usecase for public? Apart from hosting a website (which we do not have the infrastructure now). If we can cover all the below scenarios with private, then I thin we are good:
1- Keep a file to myself
2- share file with one more person
3- share the file with a group of people
4- share file via link with anyone with the link
5- revoke access to the shared file
6- share a folder

**Reply by farhoud** [reply]
7- tenancy

**Comment by farhoud** [comment]
@ehsan6sha 
The private Network will only satisfy :
- Don't put my files on the Public Network. Which i think relate to first item in requirement.

Other requirement are part of access control layer on a public network. @ruffiano89 

[simple overview](https://www.figma.com/file/zwtk0pUB3nce7UgKkLMWna/Untitled?node-id=0%3A1)
<img width="2851" alt="Untitled(1)" src="https://user-images.githubusercontent.com/2462150/171989085-961c5ddc-4f41-4525-bf83-7e788c9e9e4c.png">

**Reply by ehsan6sha** [reply]
@farhoud  hm... aren't we implementing the private network (swarm key) and adding ACL to the private network?
I mean the only difference between public and private is the swarm key, right? Then what is the downside if we add swarm key to the design? Then, If I want to share a pool, I share the swarm key. If I want to keep it totally private, I don't share swarm key

**Reply by farhoud** [reply]
@ehsan6sha 
A little note on IPFS-Cluster if we will use it as Pool base:
ipfs-cluster has its own libp2p (host) and this libp2p have have its own network that protect by cluster-secret.
Cluster has state:
File with CID of etc pinned on PeerID etc.
And ipfs-cluster based on this state pin and unpin CID on the IPFS. using RPC interface.
So we can have multiple pools controlling the same instance of IPFS.

**Reply by farhoud** [reply]
@ehsan6sha
There are other way's to create private network like:
 - We can use our Identity to create private network by whitelisting and blacklisting of peerIds.

I think we have more knowledge and help like @masih and @ruffiano89  . we should get back to the requirement and concepts and refactor our designs.

**Reply by farhoud** [reply]
Let everyone know and understand the big picture so  they could help us improve.

**Reply by ehsan6sha** [reply]
I believe what we have on non-technical big picture is the scenarios we discussed
